services/motion/builtin/builtin.go

```diff
@@ -157,32 +185,42 @@ func (ms *builtIn) MoveSingleComponent(
 	extra map[string]interface{},
 ) (bool, error) {
 	operation.CancelOtherWithLabel(ctx, "motion-service")
-	logger := ms.r.Logger()
 
-	components := robot.AllResourcesByName(ms.r, componentName.ShortName())
-	if len(components) != 1 {
-		return false, fmt.Errorf("got %d resources instead of 1 for (%s)", len(components), componentName.ShortName())
+	_, allResources, err := ms.fsService.AllCurrentInputs(ctx)
+	if err != nil {
+		return false, err
+	}
+	
+	singleComponentMoves := map[string]arm.Arm{}
+	for n, r := range allResources {
+		if thisArm, ok := r.(arm.Arm); ok {
+			singleComponentMoves[n] = thisArm
+		}
 	}
-	movableArm, ok := components[0].(arm.Arm)
+
+	movableArm, ok := singleComponentMoves[componentName.ShortName()]
 	if !ok {
-		return false, fmt.Errorf("%v(%T) is not an Arm and cannot MoveToPosition with a Pose", componentName.ShortName(), components[0])
+		return false, fmt.Errorf(
+			"could not find an arm named %v. MoveSingleComponent only supports moving arms for now",
+			componentName.ShortName(),
+		)
 	}
 
 	// get destination pose in frame of movable component
 	goalPose := destination.Pose()
 	if destination.Parent() != componentName.ShortName() {
-		logger.Debugf("goal given in frame of %q", destination.Parent())
+		ms.logger.Debugf("goal given in frame of %q", destination.Parent())
 
-		frameSys, err := framesystem.RobotFrameSystem(ctx, ms.r, worldState.Transforms)
+		frameSys, err := ms.fsService.FrameSystem(ctx, worldState.Transforms)
 		if err != nil {
 			return false, err
 		}
 		// get the initial inputs
-		fsInputs, _, err := framesystem.RobotFsCurrentInputs(ctx, ms.r, frameSys)
+		fsInputs, _, err := ms.fsService.AllCurrentInputs(ctx)
```

Comment:
[minor] Could `fsInputs` be assigned on L189 and used here to cut down on a call to `AllCurrentInputs`?